### DISCUSSION TITLE
Generating barcodes from API

### DISCUSSION THREAD

**Discussion by kerbersystems** [Budibase/budibase#9454] [Help] [answered]
Hey everyone. So, I'm trying to find out if it's possible to show the PNG result from an API call. Zebra has a free barcode generator API that returns a PNG. I can't seem to get it to display in Budibase as it can't accept that response. Could I theoretically have the API send to file to an S3 bucket and call it from there?

**Comment by melohagan** [comment, marked as answer]
Hey @kerbersystems 

This is currently not directly supported, and I have raised a feature request here: https://github.com/Budibase/budibase/issues/9461

> Could I theoretically have the API send to file to an S3 bucket and call it from there?

There could be something here! Possibly you could have an AWS lambda with the following flow:

1. Budibase makes a request to the lambda
2. The lambda reaches out the Zebra API and saves the images to a bucket.
3. Budibase retrieves the [image from an S3 bucket](https://docs.budibase.com/docs/image#displaying-images-from-an-s3-bucket)

There's a somewhat similar idea here: https://github.com/Budibase/budibase/discussions/7439

**Reply by kerbersystems** [reply]
Thank you so much! The S3 route might be better as I can "cache" the barcodes there and not have to call Zebra on every page load. It would be nice to have the ability to present images from and API response though. Thank you!
